app/autosynccontroller.cpp

```diff
@@ -8,33 +8,107 @@
  ***************************************************************************/
 
 #include "autosynccontroller.h"
+#include "coreutils.h"
 
-AutosyncController::AutosyncController( QObject *parent ) : QObject( parent )
-{
-  // raise NotImplemented :)
-}
+#include "qgsproject.h"
+#include "qgsvectorlayer.h"
 
-AutosyncController::~AutosyncController()
+AutosyncController::AutosyncController(
+  LocalProject *openedProject,
+  QgsProject *openedQgsProject,
```

Comment:
I am not completely sure without looking deeper, but do we need QgsProject as separate argument everywhere? why it is not part of LocalProject ?

Reply:
Yeah, ideally we should consider this. Currently `LocalProject` does not include `QgsProject` as you can have LocalProject instance even without it being loaded. We could have something similar to `ActiveLayer` class (maybe `ActiveProject`) that would group these and possibly other properties that are needed for opened (active) project.

Reply:
No longer valid